pallets/identity-management/src/lib.rs

```diff
@@ -227,5 +245,72 @@ pub mod pallet {
 					None => err,
 				})
 		}
+
+		// The following extrinsics are supposed to be called by TEE only
+		#[pallet::weight(195_000_000)]
+		pub fn user_shielding_key_set(
```

Comment:
In the future version those extrinsics should only be trigged by TEE-worker.

Reply:
> In the future version those extrinsics should only be trigged by TEE-worker.

yes it will be controlled by `EventTriggerOrigin`